worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -758,6 +808,76 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
     }
   }
 
+  class PushMergedDataCallback(callback: RpcResponseCallback) {
+    private val partitionIndex2StatusCode = new mutable.HashMap[Int, Byte]()
+
+    def addUnsuccessfulPartition(index: Int, statusCode: StatusCode): Unit = {
+      partitionIndex2StatusCode.put(index, statusCode.getValue)
+    }
+
+    def containIndex(index: Int): Boolean = {
+      partitionIndex2StatusCode.contains(index)
+    }
+
+    def getStatusCode(index: Int): Byte = {
+      partitionIndex2StatusCode.getOrElse(index, -1)
+    }
+
+    def unionUnsuccessfulPartition(
+        replicaPartitionIndexes: Array[Int],
+        replicaStatusCodes: Array[Byte]): Unit = {
+      if (replicaPartitionIndexes.length != replicaStatusCodes.length) {
+        throw new IllegalArgumentException(
+          "replicaPartitionIndexes and replicaStatusCodes must have the same size")
+      }
+      for (i <- replicaPartitionIndexes.indices) {
+        val index = replicaPartitionIndexes(i)
+        // if primary and replica have the same index, use the primary's status code
+        if (!partitionIndex2StatusCode.contains(index)) {
+          partitionIndex2StatusCode.put(index, replicaStatusCodes(i))
+        }
+      }
+    }
+
+    def getSortedSkipIndexes(): Array[Int] = {
```

Comment:
```suggestion
    def getFailedPartitionIndexes(): Array[Int] = {
```

Reply:
Only the partitions with "failed" or "hard_split" will be included here, as they are unwritable. Therefore, I prefer to rename it "unwritableIndexes". In addition, I will add comments to this method.